framework/src/org/checkerframework/common/value/ValueAnnotatedTypeFactory.java

```diff
@@ -171,15 +182,41 @@ public Void visitDeclared(AnnotatedDeclaredType type, Void p) {
          * If any constant-value annotation has &gt; MAX_VALUES number of values provided, treats
          * the value as UnknownVal. Works together with ValueVisitor.visitAnnotation, which issues a
          * warning to the user in this case.
+         *
+         * <p>Additional rules applied for IntVal and IntRange annotations: if anno is IntVal and
```

Comment:
IMHO, having a veeeery long IntVal annotation without replacing it with @IntRange or @UnknownVal might cause high computational complexity. I think this is why we have the MAX_VALUES in the first place. 

Besides, a veery long @IntVal would be replaced anyway because the `createIntValAnnotation` is called frequently during computations, which would replace any @IntVal with too many possible values.

Any suggestions?